Mathlib/Analysis/ODE/Gronwall.lean

```diff
@@ -126,6 +126,22 @@ theorem norm_le_gronwallBound_of_norm_deriv_right_le {f f' : ℝ → E} {δ K ε
   le_gronwallBound_of_liminf_deriv_right_le (continuous_norm.comp_continuousOn hf)
     (fun x hx _r hr => (hf' x hx).liminf_right_slope_norm_le hr) ha bound
 
+/-- Let `f : [a, b] → E` be a differentiable function such that `f a = 0`
+and `‖f'(x)‖ ≤ K ‖f(x)‖` for some constant `K`. Then `f = 0` on `[a, b]`. -/
+theorem eq_zero_of_abs_deriv_le_mul_abs_self_of_eq_zero_right
+    {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E] {f f' : ℝ → E}
+    {K a b : ℝ} (hf : ContinuousOn f (Set.Icc a b))
+    (hf' : ∀ x ∈ Set.Ico a b, HasDerivWithinAt f (f' x) (Set.Ici x) x)
+    (ha : f a = 0) (bound : ∀ x ∈ Set.Ico a b, ‖f' x‖ ≤ K * ‖f x‖) (x : ℝ) :
+    x ∈ Set.Icc a b → f x = 0 := by
+  intro hx
+  apply norm_eq_zero.mp <| le_antisymm _ (norm_nonneg (f x))
+  calc ‖f x‖ ≤ gronwallBound 0 K 0 (x - a) :=
+    norm_le_gronwallBound_of_norm_deriv_right_le hf hf' ?_ ?_ _ hx
+    _ = 0 := by rw [gronwallBound_ε0_δ0]
+  · simp [ha]
+  · simpa using bound
+
```

Comment:
I'd indent this differently, and in fact inline the last two proofs, as they are so short. Untested, but something very close to this should work.
```suggestion
  calc ‖f x‖
    _ ≤ gronwallBound 0 K 0 (x - a) :=
    norm_le_gronwallBound_of_norm_deriv_right_le hf hf' (by simp [ha]) (by simpa using bound) _ hx
    _ = 0 := by rw [gronwallBound_ε0_δ0]
```